### PR TITLE
add `D.OneHotCategorical`

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**

### CODE REVIEW

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Would you resolve the conflict?

**Comment by takagi** [comment]
`scipy.stats.multinomial` is supported only SciPy 0.19 and later, and it looks that it causes CI failure. Is it possible to use `numpy.random.multinomial` instead?

File: tests/chainer_tests/distributions_tests/test_one_hot_categorical.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+
+@testing.parameterize(*testing.product({
+    'shape': [(2, 3), ()],
+    'is_variable': [True, False],
+    'sample_shape': [(3, 2), ()],
+    'k': [3],
+}))
+@testing.fix_random()
+@testing.with_requires('scipy')
```
Comment (takagi) [inline tests/chainer_tests/distributions_tests/test_one_hot_categorical.py]: Would you apply the following change to use `scipy.stats.multinomial`?
```suggestion
@testing.with_requires('scipy>=0.19')
```

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
LGTM!

**Pull request merged**